parser/src/java/org/apache/hadoop/hive/ql/parse/AlterClauseParser.g

```diff
@@ -99,6 +100,19 @@ alterTblPartitionStatementSuffix[boolean partition]
   | alterStatementSuffixAddCol
   | alterStatementSuffixUpdateColumns
   ;
+    
+optimizeTableStatementSuffix
```

Comment:
We already have `alter table execute` syntax for partial iceberg maintance operations:
https://github.com/apache/hive/blob/94f318959ca6ec5f815e6326241bd2269ce91b80/parser/src/java/org/apache/hadoop/hive/ql/parse/AlterClauseParser.g#L473

How do we distinguish between the two syntax? 
Is the `optimize` only used for iceberg compaction?

I'm not opposed to adding new syntax, but i think we need to clarify the specific use cases to avoid users confusion.

cc @ayushtkn @deniskuzZ any thoughts?

Reply:
Optimize syntax as proposed in this PR is a syntactic sugar for existing compaction syntax in Hive and it works for both Iceberg and ACID tables, but it is meant for Iceberg tables and the reason is that many Hive competitors use Optimize syntax for Iceberg compaction and it may help with Hive Iceberg adoption.
We are planning to extend the support for Optimize syntax in the future to support parameters like minimum number of files in the partition and file size threshold parameters which will be only for Iceberg:

```
OPTIMIZE TABLE <table_name>
REWRITE DATA [USING BIN_PACK]
[ ( { FILE_SIZE_THRESHOLD | MIN_INPUT_FILES } = <value> [, ... ] ) ]
WHERE category = 'c1'
```

Reply:
what is bad in our current syntax? we use that for acid, we will use it for iceberg as well, syntatic sugar makes sense when things go easy.

why don't the competitor use our syntax :-) 

no objections though from my side, btw. is it coming from spark or impala?

Reply:
I think it is fine if the big data world has common syntax. It provides an easier learning curve.

Reply:
looks good to me